warp10/src/main/java/io/warp10/script/functions/FIND.java

```diff
@@ -317,14 +368,30 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         drequest.setLabelsSelectors(lblsSels);
       } else {
         // Fix labels
-        
+
         if (drequest.isSetLabelsSelectors()) {
-          for (Map<String,String> sel: drequest.getLabelsSelectors()) {
+          for (int i = 0; i < drequest.getLabelsSelectorsSize(); i++) {
+            Map<String,String> sel = drequest.getLabelsSelectors().get(i);
             sel.remove(Constants.PRODUCER_LABEL);
             sel.remove(Constants.OWNER_LABEL);
             sel.remove(Constants.APPLICATION_LABEL);
             sel.putAll(Tokens.labelSelectorsFromReadToken(rtoken));
-          }          
+
+            // Re-order the labels
+            Map<String,String> ordered = new LinkedHashMap<String,String>(null != labelSelectors ? labelSelectors.size() : 1);
```

Comment:
I think it should be `Map<String,String> ordered = new LinkedHashMap<String,String>(sel.size());`.